File: articles/institutional-path-dependence/fortran/path_dependence_score.f90
program path_dependence_score
  implicit none

  real :: initial_conditions, behavioral_reinforcement, feedback_strength
  real :: increasing_returns, coordination_effects, learning_effects
  real :: legitimacy, switching_costs, complementarity
  real :: disruption_pressure, reform_capacity
  real :: score

  initial_conditions = 80.0
  behavioral_reinforcement = 75.0
  feedback_strength = 70.0
  increasing_returns = 82.0
  coordination_effects = 78.0
  learning_effects = 74.0
  legitimacy = 85.0
  switching_costs = 88.0
  complementarity = 79.0
  disruption_pressure = 35.0
  reform_capacity = 40.0

  score = 0.08 * initial_conditions + &
          0.12 * behavioral_reinforcement + &
          0.12 * feedback_strength + &
          0.13 * increasing_returns + &
          0.11 * coordination_effects + &
          0.10 * learning_effects + &
          0.12 * legitimacy + &
          0.12 * switching_costs + &
          0.10 * complementarity - &
          0.12 * disruption_pressure - &
          0.05 * reform_capacity

  print *, "Institutional path dependence raw score:", score
end program path_dependence_score

File: articles/institutional-path-dependence/cpp/path_dependence_score.cpp
#include <iostream>

struct PathDependenceCase {
    double initial_conditions;
    double behavioral_reinforcement;
    double feedback_strength;
    double increasing_returns;
    double coordination_effects;
    double learning_effects;
    double legitimacy;
    double switching_costs;
    double complementarity;
    double disruption_pressure;
    double reform_capacity;
};

double path_dependence_score_raw(const PathDependenceCase& x) {
    return
        0.08 * x.initial_conditions +
        0.12 * x.behavioral_reinforcement +
        0.12 * x.feedback_strength +
        0.13 * x.increasing_returns +
        0.11 * x.coordination_effects +
        0.10 * x.learning_effects +
        0.12 * x.legitimacy +
        0.12 * x.switching_costs +
        0.10 * x.complementarity -
        0.12 * x.disruption_pressure -
        0.05 * x.reform_capacity;
}

int main() {
    PathDependenceCase demo {80, 75, 70, 82, 78, 74, 85, 88, 79, 35, 40};
    std::cout << "Institutional path dependence raw score: "
              << path_dependence_score_raw(demo) << std::endl;
    return 0;
}
